### ISSUE TITLE
Single header file release

### ISSUE THREAD

**Issue by BrendanSimon** [d1vanov/Simple-FFT#2]
I know this project hasn't been active for a while, but if anyone is listening, I was wondering how hard it would be to geneate releases that are a **single header file** ?

An example header only project is Catch2.  I think it is multiple headers in it's implementation, but is released as a single header file for easy use and inclusion in projects.  I can easily upgrade to a new release by updating one header file.

https://github.com/catchorg/Catch2

Would be nice if Simple-FFT had a similar one header file release :)

**Comment by d1vanov** [comment]
Hi. I am listening but frankly I am not much interested in putting effort into this idea. I looked into what Catch2 project does to create a single header file and that looks kinda scary - some Python [script](https://github.com/catchorg/Catch2/blob/master/scripts/generateSingleHeader.py) parsing the C++ header files seems to be involved, so it's not something as simple as "compose these headers using cat".

I can leave this issue open for a while and see if anyone has any idea how this idea could be implemented in a simpler way and would actually like to do that.

**Comment by BrendanSimon** [comment]
Fair enough too.  If it could be made as easy as a "cat this an that", maybe with little refactoring or something, then it might be worth the effort.
